### PR TITLE
ci: add release-vm-console-proxy workflow

### PR DESCRIPTION
**Pull request by codingben**
**What this PR does / why we need it**:

This will add release vm-console-proxy workflow. This workflow will be triggered by another release workflow in vm-console-proxy repository.

**Special notes for your reviewer**:

It should be merged before we merge https://github.com/kubevirt/vm-console-proxy/pull/2.

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by codingben** [comment]
FYI @akrejcir

File: .github/workflows/release-vm-console-proxy.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      - run: |
+          RELEASE_VERSION=${{ github.event.client_payload.release_version }}
+          OUTPUT_FILE=./data/vm-console-proxy-bundle/vm-console-proxy-${RELEASE_VERSION}.yaml
+          mkdir -p ./data/vm-console-proxy-bundle
+          curl -L https://github.com/kubevirt/vm-console-proxy/releases/download/${RELEASE_VERSION}/vm-console-proxy.yaml > ${OUTPUT_FILE}
+      
+      - name: Commit generated release manifests
+        uses: EndBug/add-and-commit@v9
+        with:
+          message: "chore: update kubernetes manifests to ${{ github.event.client_payload.release_version }}"
```
Comment (0xFelix) [inline .github/workflows/release-vm-console-proxy.yaml:25]: Be more specific by writing `update vm-console-proxy to ...`?

File: .github/workflows/release-vm-console-proxy.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        with:
+          source_branch: "vm-console-proxy-${{ github.event.client_payload.release_version }}"
+          destination_branch: "main"
+          pr_title: "Update vm-console-proxy manifests to ${{ github.event.client_payload.release_version }}"
+          pr_body: |
+            Update vm-console-proxy manifests to version ${RELEASE_VERSION}
+
+            **Release note**:
+            \`\`\`release-note
+            None
```
Comment (0xFelix) [inline .github/workflows/release-vm-console-proxy.yaml:41]: I think you should add a release note.

File: .github/workflows/release-vm-console-proxy.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    name: Release VM Console Proxy
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          release_version: ${{ github.event.client_payload.release_version }}
+      - run: |
+          RELEASE_VERSION=${{ github.event.client_payload.release_version }}
+          OUTPUT_FILE=./data/vm-console-proxy-bundle/vm-console-proxy-${RELEASE_VERSION}.yaml
```
Comment (akrejcir) [inline .github/workflows/release-vm-console-proxy.yaml:18]: No need to add release version to the filename. There will be only one file.

File: .github/workflows/release-vm-console-proxy.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+          curl -L https://github.com/kubevirt/vm-console-proxy/releases/download/${RELEASE_VERSION}/vm-console-proxy.yaml > ${OUTPUT_FILE}
+
+      - name: Create pull request
+        uses: peter-evans/create-pull-request@v4
+        with:
+          release_version: ${{ github.event.client_payload.release_version }}
+          token: ${{ secrets.GH_ACCESS_TOKEN }}
+          commit-message: "chore: update kubernetes manifests to ${{ github.event.client_payload.release_version }}"
+          committer: Ben Oukhanov <boukhanov@redhat.com>
+          author: Ben Oukhanov <boukhanov@redhat.com>
```
Comment (akrejcir) [inline .github/workflows/release-vm-console-proxy.yaml:29]: can you use some bot name and nonexisting email? Like `vm-console-proxy update`.

File: .github/workflows/release-vm-console-proxy.yaml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+jobs:
+  release-vm-console-proxy:
+    name: Release VM Console Proxy
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          release_version: ${{ github.event.client_payload.release_version }}
```
Comment (akrejcir) [inline .github/workflows/release-vm-console-proxy.yaml:15]: `release_version` is not an attribute of `action/checkout`: https://github.com/actions/checkout

File: .github/workflows/release-vm-console-proxy.yaml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      - run: |
+          RELEASE_VERSION=${{ github.event.client_payload.release_version }}
+          OUTPUT_FILE=./data/vm-console-proxy-bundle/vm-console-proxy-${RELEASE_VERSION}.yaml
+          mkdir -p ./data/vm-console-proxy-bundle
+          curl -L https://github.com/kubevirt/vm-console-proxy/releases/download/${RELEASE_VERSION}/vm-console-proxy.yaml > ${OUTPUT_FILE}
+
+      - name: Create pull request
+        uses: peter-evans/create-pull-request@v4
+        with:
+          release_version: ${{ github.event.client_payload.release_version }}
```
Comment (akrejcir) [inline .github/workflows/release-vm-console-proxy.yaml:25]: Similarly, `release_version` is not an attribute of this action.

File: .github/workflows/release-vm-console-proxy.yaml (hunk 4 above)
Comment (codingben) [reply .github/workflows/release-vm-console-proxy.yaml]: Done, we're using GitHub Action bot.

File: .github/workflows/release-vm-console-proxy.yaml (hunk 6 above)
Comment (codingben) [reply .github/workflows/release-vm-console-proxy.yaml]: Yep, already saw it.

File: .github/workflows/release-vm-console-proxy.yaml (hunk 3 above)
Comment (codingben) [reply .github/workflows/release-vm-console-proxy.yaml:18]: Okay, but it's similar to common-templates [1] and kubevirt-tekton-tasks-operator [2].

[1] https://github.com/kubevirt/ssp-operator/tree/master/data/common-templates-bundle
[2] https://github.com/kubevirt/tekton-tasks-operator/tree/main/data/tekton-tasks/kubernetes

File: .github/workflows/release-vm-console-proxy.yaml (hunk 5 above)
Comment (codingben) [reply .github/workflows/release-vm-console-proxy.yaml:15]: I saw an example of it [1]. It doesn't mean that it's doing checkout code when there is an actual `release_version` passed?

File: .github/workflows/release-vm-console-proxy.yaml (hunk 2 above)
Comment (codingben) [reply .github/workflows/release-vm-console-proxy.yaml]: Yep, thanks!

File: .github/workflows/release-vm-console-proxy.yaml (hunk 3 above)
Comment (lyarwood) [reply .github/workflows/release-vm-console-proxy.yaml:23]: FWIW we wanted to move away from including all of the versioned manifests in the operator, that's why `common-instancetypes` just has one now.

**Review by lyarwood — CHANGES_REQUESTED** [review]
Like the approach of having the task in this repo, I might copy this for common-instancetypes. Just a quick nit with the version in the filename but otherwise LGTM.

File: .github/workflows/release-vm-console-proxy.yaml (hunk 3 above)
Comment (codingben) [reply .github/workflows/release-vm-console-proxy.yaml:23]: Ack. I'll remove it now.

**Comment by akrejcir** [comment]
Looks good. 
@ksimon1 Can you check if workflows in this repo have write permissions, so it can create the branch needed for PR? I don't have permissions to check it.

**Comment by akrejcir** [comment]
Can you write more details into the commit message?

**Comment by codingben** [comment]
@lyarwood @akrejcir Thanks! I just tested it in my forked repository and it works [1].

[1] https://github.com/codingben/ssp-operator/pull/3

**Comment by codingben** [comment]
> Looks good. @ksimon1 Can you check if workflows in this repo have write permissions, so it can create the branch needed for PR? I don't have permissions to check it.

@ksimon1 I guess it also need to have "Allow GitHub Actions to create and approve pull requests" checked. For example:

image

**Comment by ksimon1** [comment]
@codingben, @akrejcir permissions should be correctly set.

**Comment by akrejcir** [comment]
Thanks!

/lgtm
/approve

**Pull request merged**